OneCommunityGlobal/HighestGoodNetworkApp
# Roberto - enhances save changes modal to notify request is still processing

# Description
When making changes to a users's profile and clicking on save changes the success modal immediately pops up even though the put request isn't fully complete processing. If you make a change to the user and click save changes, then refresh your page immediately it would lead to the frontend sending out the put request, but not completely finishing due to the page refresh which would lead to a get request while the put is still being carried out. This would lead to the get request sending old information rather than the newly updated information from the put request.

## Related PRS (if any):
Use latest backend dev branch

## Main changes explained:
- In UserProfile.jsx changed the handleSubmit props of the SaveButton component to be async.
- In SaveButton.jsx added a still waiting message and updated the handleSave function to be async and trigger a change in modal message when the request is done.
- In EditConfirmModal.jsx added a boolean disabled props to make the exit button disabled in the short period that the changes are fully saved.

## How to test:
1. check into current branch
2. do `npm install` and `npm run start:local` to run this PR locally
3. log as admin user
4. go to any user profile page→ edit any basic information → click on save changes
5. verify you see the still waiting message then it turns into the success message 

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/117053202/708a059b-edb4-4359-90dc-ba34a817adbb

## Note
The message is just to prevent people from refreshing the page before the request is fully executed
